### DISCUSSION TITLE
CSRF protection for open API endpoints

### DISCUSSION THREAD

**Discussion by sensedrive** [directus/directus#9128] [Archive]
Hello everyone,

I wonder if there is a CSRF protection layer inside Directus, which creates sessions for also anonymous users. I just wanted to protect some custom open API endpoints, which add data inside the CMS. These  request should be only accepted, if a session-cookie was created and equals the current user session.

How do you handle this?

Thanks in advance!
Martin

**Comment by rijkvanzanten** [comment]
The API currently doesn't generate any sessions for non-authenticated users
